extensions/amp-analytics/0.1/test/test-requests.js

```diff
@@ -58,6 +58,103 @@ describes.realWin('Requests', {amp: 1}, env => {
+        const r = {'baseUrl': '/r1', 'origin': 'http://example.com/'};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('http://example.com/r1');
+      });
+
+      it('handle trailing path in request origin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': 'http://example.com/test'};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('http://example.com/r1');
+      });
+
+      it('handle empty requestOrigin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': ''};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('/r1');
+      });
+
+      it('handle undefined requestOrigin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': undefined};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('/r1');
+      });
+
+      it('prepend request origin to absolute baseUrl', function*() {
+        const r = {
+          'baseUrl': 'https://baseurl.com',
+          'origin': 'https://requestorigin.com',
+        };
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith(
+          'https://requestorigin.comhttps://baseurl.com'
+        );
+      });
+
+      it('should expand request origin', function*() {
+        const r = {'baseUrl': '/r2', 'origin': '${documentReferrer}'};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({
+          'documentReferrer': 'http://example.com',
+        });
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('http://example.com/r2');
+      });
+
+      it('should expand nested request origin', function*() {
+        const r = {'baseUrl': '/r3', 'origin': '${a}'};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({
+          'a': '${b}',
+          'b': 'http://example.com',
+        });
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('http://example.com/r3');
+      });
+    });
```

Comment:
We talked about this, but I still think it's worth adding the `requestOrigin` being relative url test. Tests are mostly for developers, and it helps them to understand the desired behavior much better. With that test, we could save a lot of explanation on how we want to handle the special case where `requestOrigin` is not absolute. WDYT?